src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -24,16 +24,100 @@
 #include "third_party/htmlDoc/htmlDoc.hpp"
 #include "third_party/progress/progress.hpp"
 
+#ifdef HAVE_BOOST
+#include <boost/format.hpp>
+#endif
+
+#include <tuple>
+
 #ifdef _MSC_VER
 #pragma warning( once : 4267 ) //warning C4267: 'argument' : conversion from 'size_t' to 'const int', possible loss of data
 #endif
 
+//#define OPENMVG_NEXTBESTVIEW_WITHOUT_SCORE
+
 namespace openMVG {
 namespace sfm {
 
 using namespace openMVG::geometry;
 using namespace openMVG::cameras;
 
+/**
+ * @brief Compute indexes of all features in a fixed size pyramid grid.
+ * These precomputed values are useful to the next best view for incremental SfM.
+ *
+ * @param[in] tracksPerView: The list of TrackID per view
+ * @param[in] map_tracks: All putative tracks
+ * @param[in] views: All views
+ * @param[in] featuresProvider: Input features and descriptors
+ * @param[in] pyramidDepth: Depth of the pyramid.
+ * @param[out] tracksPyramidPerView:
+ *             Precomputed list of pyramid cells ID for each track in each view.
+ */
+void computeTracksPyramidPerView(
+    const tracks::TracksPerView& tracksPerView,
+    const tracks::STLMAPTracks& map_tracks,
+    const Views& views,
+    const Features_Provider& featuresProvider,
+    const std::size_t pyramidDepth,
+    tracks::TracksPyramidPerView& tracksPyramidPerView)
+{
+  std::vector<std::size_t> widthPerLevel(pyramidDepth);
+  std::vector<std::size_t> startPerLevel(pyramidDepth);
+  std::size_t start = 0;
+  for(size_t level = 0; level < pyramidDepth; ++level)
+  {
+    startPerLevel[level] = start;
+    widthPerLevel[level] = std::pow(2.0, level+1);
+    start += Square(widthPerLevel[level]);
+  }
+
+#ifdef HAVE_BOOST
+  tracksPyramidPerView.reserve(tracksPerView.size());
+#endif
+  for(const auto& viewTracks: tracksPerView)
+  {
+    auto& trackPyramid = tracksPyramidPerView[viewTracks.first];
+#ifdef HAVE_BOOST
```

Comment:
it could be just a single `#ifdef HAVE_BOOST`, no? It's useless to do the loop if there is nothing to do.